Utilities/DataFlow/test/test_TimeframeParser.cxx

```diff
@@ -0,0 +1,55 @@
+#include "DataFlow/TimeframeParser.h"
+#include "Headers/DataHeader.h"
+#include <FairMQParts.h>
+#include <istream>
+#include <cstdlib>
+
+
+struct OneShotReadBuf : public std::streambuf
+{
+    OneShotReadBuf(char* s, std::size_t n)
+    {
+        setg(s, s, s + n);
+    }
+};
+
+using DataHeader = o2::Header::DataHeader;
+
+int
+main(int argc, char **argv) {
+
+  // Construct a dummy timeframe.
+  // Stream it and get the parts
+  FairMQParts parts;
+  auto onAddParts = [](FairMQParts &p, char *buffer, size_t size) {
+    LOG(INFO) << "Adding part to those to be sent.\n";
+  };
+  auto onSend = [](FairMQParts &p) {
+    LOG(INFO) << "Sending parts\n";
+  };
+  auto testBufferSize = sizeof(o2::Header::DataHeader) + 1000 + sizeof(o2::Header::DataHeader) + sizeof(o2::Header::DataHeader);
+  char *testBuffer = new char[testBufferSize];
+  auto dh = reinterpret_cast<DataHeader*>(testBuffer);
+  auto idh = reinterpret_cast<DataHeader*>(testBuffer + 1000);
+  // FIXME: Do we really expect that dataDescription is set by the sender?
```

Comment:
if you mean setting the size: the default ctor does that for you - essentially all the members inherited from BaseHeader are set by the default ctor - they should not be touched by user code. Although since this is test code we might want to explicitly check if they are set correctly.
That being said, probably here you want to construct the header in place in the buffer you allocated.
Now the actual interface to construct buffers that hold metadata is the Stack class which packs a buffer with all the headers you might want to have in the header stack - it is a move only type so it plays nice with the transport layer. I need to clean this up a bit and finish separating the public and private interfaces - e.g. we should not initialize the description member with an arbitrary string - this is in the works.

example of how it was meant to be used is in Utilities/O2MessageMonitor.cxx

so in a test you could do something like:
DataHeader dataHeader;
dataHeader = gDataOriginAny; //set origin
dataHeader = gDataDescriptionInfo; //set data description
dataHeader = gSerializationMethodNone; //set serialization method for the payload

o2::Header::Stack headerStack{dataHeader};

byte* buffer = headerStack.data();
size_t size = headerStack.size();

//check the metadata buffer byte by byte somehow....